File: C#/find-the-pivot-integer.cs
public class Solution {
    public int PivotInteger(int n) {
        if (n == 1) {
            return 1;
        }
        int total = (1 + n) * n / 2;
        int mid = n / 2;        
        for (int x = mid; x < n; x++) {
            int low = (1 + x) * x / 2;
            int high = total - low + x;
            if (low == high) {
                return x;
            } else if (low > high) {
                break;
            }
        }
        return -1;
    }
}

File: Go/find-the-pivot-integer.go
func pivotInteger(n int) int {
	total := (1 + n) * n / 2
	//left = right
	//right = total - left + x
	//(1 + x) * x / 2 = (1+n)*n/2 - (1+x)*x/2 + x
	//x + x*x = n + n*n - x - x*x + x*2
	//x = sqrt((n + n*n) / 2)
    x := int(math.Sqrt(float64(n + n*n) / 2))
    if x * x == total {
        return x
    }
	return -1
}